### DISCUSSION TITLE
How to OTA update my app on ESP32?

### DISCUSSION THREAD

**Discussion by jeffer80** [nanoframework/Home#1276] [Q&A] [answered]
I think the main steps are:
1. Check for new versions via my Cloud API
2. Download the bin file and save it to flash
3. Rewrite the app partition and restart

My question is how to re-write the application at runtime.
espressif provided a solution under IDF, now what should I do in nanoCLR?
https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/system/ota.html

**Comment by josesimoes** [comment, marked as answer]
The in-field update is not available yet.
Until that happens the closest you get is using reflection to load an assembly through reflection.

https://github.com/nanoframework/Samples/tree/main/samples/AzureSDK/AzureEdgeOta

**Reply by jeffer80** [reply]
That's mean I run a main process like bootloader,
then load my app as assembly,
If I want OTA my app, just download the assembly file and reload it 

that's a good idea, I'll try it. thx

**Reply by jonas-vanhoof01** [reply]
Hey, did you manage to get it working? I'm considering to use nanoframework for my project and OTA is just an important decision point for me.
